internal/controllers/machine/machine_controller_noderef_test.go

```diff
@@ -253,3 +257,79 @@ func TestGetManagedLabels(t *testing.T) {
 	got := getManagedLabels(allLabels)
 	g.Expect(got).To(BeEquivalentTo(managedLabels))
 }
+
+func TestReconcileNodeTaints(t *testing.T) {
```

Comment:
I think it's fine to just unit test with the fake client here. In my opinion no need to migrate this to envtest